pkg/generator/deploy_tmpl.go

```diff
@@ -33,19 +33,23 @@ const operatorYamlTmpl = `apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{.ProjectName}}
+  labels:
+    k8s-app: {{.ProjectName}}
 spec:
   replicas: 1
   selector:
     matchLabels:
-      name: {{.ProjectName}}
+      k8s-app: {{.ProjectName}}
   template:
     metadata:
       labels:
-        name: {{.ProjectName}}
+        k8s-app: {{.ProjectName}}
     spec:
       containers:
         - name: {{.ProjectName}}
           image: {{.Image}}
+          ports:
+          - containerPort: 9090
```

Comment:
can we `name` this port? and then use that in the service as `targetPort: name`.  That leaves us flexibility to change the port in the future without having to change the service.

Reply:
Sure ! 👍

Reply:
containerPort is now named `metrics` and use in the service as `targetPort: metrics`

Reply:
Make the port number a variable like `{{.MetricsPort}}` and define a new parameter in [gen_deploy.go](https://github.com/operator-framework/operator-sdk/blob/master/pkg/generator/gen_deploy.go#L60-L63)
```Go
type OperatorYaml struct {
	ProjectName string
	Image       string
	MetricsPort int
}
```
That way we can set MetricsPort to a constant defined elsewhere.